### PR TITLE
Add Dockerfile example with npm build script

### PR DESCRIPTION
**Pull request by dmp1ce**
Using the [a guide from the README](https://github.com/dangeross/guides/blob/master/raspibolt/raspibolt_6B_lndhub.md), I thought it might be helpful to have a simple Dockerfile example.

I also added a npm build script to simplify building the project and allow for using the start script in production. I used [this production guide](https://github.com/babel/example-node-server#getting-ready-for-production-use) for reference.

### CODE REVIEW

**Comment by Overtorment** [comment]
thanks!

**Comment by Overtorment** [comment]
heroku is having problem building this. do you know how to fix it?


```
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       NODE_VERBOSE=false
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version 12.x...
       Downloading and installing node 12.20.0...
       Using default npm version: 6.14.8
       
-----> Restoring cache
       - node_modules
       
-----> Installing dependencies
       Installing node modules
       
       > grpc@1.24.4 install /tmp/build_120cfac8_/node_modules/grpc
       > node-pre-gyp install --fallback-to-build --library=static_library
       
       [grpc] Success: "/tmp/build_120cfac8_/node_modules/grpc/src/node/extension_binary/node-v72-linux-x64-glibc/grpc_node.node" is installed via remote
       
       > secp256k1@3.8.0 install /tmp/build_120cfac8_/node_modules/secp256k1
       > npm run rebuild || echo "Secp256k1 bindings compilation fail. Pure JS implementation will be used."
       
       
       > secp256k1@3.8.0 rebuild /tmp/build_120cfac8_/node_modules/secp256k1
       > node-gyp rebuild
       
       make: Entering directory '/tmp/build_120cfac8_/node_modules/secp256k1/build'
         CXX(target) Release/obj.target/secp256k1/src/addon.o
         CXX(target) Release/obj.target/secp256k1/src/privatekey.o
       ../src/privatekey.cc: In function ‘Nan::NAN_METHOD_RETURN_TYPE privateKeyNegate(Nan::NAN_METHOD_ARGS_TYPE)’:
       ../src/privatekey.cc:73:30: warning: ignoring return value of ‘int secp256k1_ec_privkey_negate(const secp256k1_context*, unsigned char*)’, declared with attribute warn_unused_result [-Wunused-result]
          secp256k1_ec_privkey_negate(secp256k1ctx, &private_key[0]);
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         CXX(target) Release/obj.target/secp256k1/src/publickey.o
         CXX(target) Release/obj.target/secp256k1/src/signature.o
         CXX(target) Release/obj.target/secp256k1/src/ecdsa.o
       ../src/ecdsa.cc: In function ‘Nan::NAN_METHOD_RETURN_TYPE sign(Nan::NAN_METHOD_ARGS_TYPE)’:
       ../src/ecdsa.cc:88:131: warning: ignoring return value of ‘v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)’, declared with attribute warn_unused_result [-Wunused-result]
          obj->Set(info.GetIsolate()->GetCurrentContext(), Nan::New<v8::String>("signature").ToLocalChecked(), COPY_BUFFER(&output[0], 64));
                                                                                                                                          ^
       In file included from /app/.cache/node-gyp/12.20.0/include/node/node.h:67:0,
                        from ../src/ecdsa.cc:1:
       /app/.cache/node-gyp/12.20.0/include/node/v8.h:3504:37: note: declared here
          V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                            ^~~
       ../src/ecdsa.cc:89:130: warning: ignoring return value of ‘v8::Maybe<bool> v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>, v8::Local<v8::Value>)’, declared with attribute warn_unused_result [-Wunused-result]
          obj->Set(info.GetIsolate()->GetCurrentContext(), Nan::New<v8::String>("recovery").ToLocalChecked(), Nan::New<v8::Number>(recid));
                                                                                                                                         ^
       In file included from /app/.cache/node-gyp/12.20.0/include/node/node.h:67:0,
                        from ../src/ecdsa.cc:1:
       /app/.cache/node-gyp/12.20.0/include/node/v8.h:3504:37: note: declared here
          V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                            ^~~
         CXX(target) Release/obj.target/secp256k1/src/ecdh.o
         CC(target) Release/obj.target/secp256k1/src/secp256k1-src/src/secp256k1.o
         CC(target) Release/obj.target/secp256k1/src/secp256k1-src/contrib/lax_der_parsing.o
         CC(target) Release/obj.target/secp256k1/src/secp256k1-src/contrib/lax_der_privatekey_parsing.o
         SOLINK_MODULE(target) Release/obj.target/secp256k1.node
         COPY Release/secp256k1.node
       make: Leaving directory '/tmp/build_120cfac8_/node_modules/secp256k1/build'
       
       > core-js@2.6.11 postinstall /tmp/build_120cfac8_/node_modules/core-js
       > node -e "try{require('./postinstall')}catch(e){}"
       
       
       > core-js@3.8.0 postinstall /tmp/build_120cfac8_/node_modules/@babel/node/node_modules/core-js
       > node -e "try{require('./postinstall')}catch(e){}"
       
       added 972 packages in 28.793s
       
-----> Build
       Running build
       
       > LndHub@1.2.1 build /tmp/build_120cfac8_
       > babel ./ -d dist --copy-files --ignore node_modules
       
SyntaxError: /tmp/build_120cfac8_/.heroku/node/lib/node_modules/npm/docs/gatsby-browser.js: Support for the experimental syntax 'jsx' isn't currently enabled (7:10):
  5 | 
  6 | export const wrapPageElement = ({ element, props }) => {
> 7 |   return <Layout {...props} >{element}</Layout>
    |          ^
  8 | }
  9 | 
Add @babel/preset-react (https://git.io/JfeDR) to the 'presets' section of your Babel config to enable transformation.
If you want to leave it as-is, add @babel/plugin-syntax-jsx (https://git.io/vb4yA) to the 'plugins' section to enable parsing.
    at Parser._raise (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:748:17)
    at Parser.raiseWithData (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:741:17)
    at Parser.expectOnePlugin (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:9112:18)
    at Parser.parseExprAtom (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:10542:22)
    at Parser.parseExprSubscripts (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:10122:23)
    at Parser.parseUpdate (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:10102:21)
    at Parser.parseMaybeUnary (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:10091:17)
    at Parser.parseExprOps (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:9961:23)
    at Parser.parseMaybeConditional (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:9935:23)
    at Parser.parseMaybeAssign (/tmp/build_120cfac8_/node_modules/@babel/core/node_modules/@babel/parser/lib/index.js:9898:21) {
  loc: Position { line: 7, column: 9 },
  pos: 209,
  missingPlugin: [ 'jsx', 'flow', 'typescript' ],
  code: 'BABEL_PARSE_ERROR'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! LndHub@1.2.1 build: `babel ./ -d dist --copy-files --ignore node_modules`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the LndHub@1.2.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/npmcache.4ViVG/_logs/2020-12-08T13_23_31_985Z-debug.log
-----> Build failed
       
       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys
       
       Some possible problems:
       
       - Node version not specified in package.json
         https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
       
       Love,
       Heroku
       
 !     Push rejected, failed to compile Node.js app.
 !     Push failed
```

**Comment by dmp1ce** [comment]
Unfortunately I didn't test on Heroku. Did 54425c51c821fa7df303644b1306f109cf664822 resolve the issue?

**Comment by Overtorment** [comment]
> Unfortunately I didn't test on Heroku. Did [54425c5](https://github.com/BlueWallet/LndHub/commit/54425c51c821fa7df303644b1306f109cf664822) resolve the issue?

no

**Comment by dmp1ce** [comment]
I'm looking into it. It looks like this PR was only half-baked.

**Comment by dmp1ce** [comment]
Maybe it is best to roll back to when Heroku worked. I'm trying to find a fix [here](https://github.com/dmp1ce/LndHub/tree/fix-build) and testing on Heroku but I haven't been able to find a solution yet.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "dev": "nodemon node_modules/.bin/babel-node index.js",
-    "start": "node_modules/.bin/babel-node index.js",
+    "build": "babel ./ -d dist --copy-files --ignore node_modules",
```
Comment (dmp1ce) [inline package.json:9]: This line needed to ignore node_modules with `\"**node_modules\*` in order to work on Heroku.

Comment (Overtorment) [reply package.json:9]: I didnt get it.
if you solved it - redo the PR

**Comment by dmp1ce** [comment]
I think I found the issue on Heroku. The `node_modules` directory wasn't being ignored like it should have been.

1. I am going to rework the `Dockerfile` so there are no changes needed for `package.json`.
2. I'm also going to work on a PR for moving the javascript source code into the `src` directory. This would make the builds using babel simpler.

File: package.json (hunk 1 above)
Comment (dmp1ce) [reply package.json:9]: The PR is not ready yet. I just wanted to let you know what the issue was, if you were curious.

**Pull request merged**